### PR TITLE
Update preview branch to use artifacts with "-preview" suffix

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Also updating device client API version to preview, for digital twin capabilities

### CODE REVIEW

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceTwin/Property.java [hunk 1]
```diff
@@ -23,13 +23,6 @@ public Property(String key, Object value)
              */
             throw new IllegalArgumentException("Key cannot be null or empty");
         }
-        if (key.contains(" ") || key.contains("$") || key.contains("."))
-        {
-            /*
-            **Codes_SRS_Property_25_006: [**If the key contains illegal unicode control characters i.e ' ', '.', '$', the constructor shall throw an IllegalArgumentException.**]**
-             */
-            throw new IllegalArgumentException("Key cannot contain illegal unicode control characters '.', '$', ' '");
```
Comment (prmathur-microsoft) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceTwin/Property.java]: Can you split this commit into separate one. Please follow github hygiene for meaningful commits.

File: device/iot-device-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/device/DeviceTwin/PropertyTest.java [hunk 2]
```diff
@@ -68,33 +68,6 @@ public void constructorWithMetadataThrowsOnNullKey()
...
-        Property testProp = new Property("Key with space", 1);
-
-    }
-
-    @Test (expected = IllegalArgumentException.class)
-    public void constructorThrowsOnInvalidKey_2()
-    {
-        //act
-        Property testProp = new Property("KeyWith$", 1);
-
```
Comment (prmathur-microsoft) [inline device/iot-device-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/device/DeviceTwin/PropertyTest.java]: Also shoudn't this commit go on master? Why is it deviating from master ?

File: provisioning/security/dice-provider/pom.xml [hunk 3]
```diff
@@ -5,11 +5,16 @@
...
+    <parent>
+        <groupId>com.microsoft.azure.sdk.iot.provisioning.security</groupId>
+        <artifactId>security</artifactId>
+        <version>1.3.0</version>
+    </parent>
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.microsoft.azure.sdk.iot.provisioning.security</groupId>
-    <artifactId>dice-provider</artifactId>
+    <artifactId>dice-provider-preview</artifactId>
     <name>Provisioning Security DICE Provider</name>
```
Comment (prmathur-microsoft) [inline provisioning/security/dice-provider/pom.xml]: Do you need this to be in preview? If so then please translate all packages to preview. I noticed some aren't in preview state

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/Pair.java [hunk 4]
```diff
@@ -21,13 +21,6 @@ public Pair(String key, Object value) throws IllegalArgumentException
              */
             throw new IllegalArgumentException("Key cannot be null or empty");
         }
-        if (key.contains(" ") || key.contains("$") || key.contains(".") || key.length() > MAX_ALLOWABLE_KEY_LENGTH)
-        {
-            /*
```
Comment (prmathur-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/Pair.java]: Please submit a separate PR on master and merge it back into preview to maintain commit consistency.

File: service/iot-service-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/service/devicetwin/PairTest.java [hunk 5]
```diff
@@ -47,37 +47,6 @@ public void constructorThrowsIfKeyEmpty()
...
-    /*
-    **Tests_SRS_Pair_25_003: [**If the key contains illegal unicode control characters i.e ' ', '.', '$', the constructor shall throw an IllegalArgumentException.**]**
-     */
-    @Test (expected = IllegalArgumentException.class)
-    public void constructorThrowsIfKeyIllegalCharacters_1()
-    {
-        //act
-        Pair testPair = new Pair("Key With Space", "TestObject");
-    }
-
```
Comment (prmathur-microsoft) [inline service/iot-service-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/service/devicetwin/PairTest.java]: Separate PR please.

File: versions.json [hunk 6]
```diff
@@ -1,15 +1,13 @@
...
-    "securityProvider": "1.2.0",
-    "tpmEmulator": "1.1.1",
-    "tpmHsm": "1.1.1",
-    "diceEmulator": "1.1.1",
-    "dice": "1.1.1",
-    "x509": "1.1.2",
-    "provisioning": "1.7.0",
-    "provisioningDevice": "1.6.0",
-    "provisioningService": "1.5.0",
-    "java": "0.24.0"
```
Comment (prmathur-microsoft) [inline versions.json]: What would  "java": "0.24.0", "provisioning": "1.7.0", follow incase of bump version only on preview? For consistency I suggest to add these as well to versions.json on preview branch

Comment (davilu) [reply versions.json]: `"java": "0.24.0", "provisioning": "1.7.0"` is parent version. They are not been used and shouldn't be changed. I'm not sure what they were used for but it should always be `1.0.0` to me.

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceTwin/Property.java (hunk 1 above)
Comment (timtay-microsoft) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceTwin/Property.java]: Sure, now this PR is just about changing artifact id's to add "-preview" suffix

File: device/iot-device-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/device/DeviceTwin/PropertyTest.java (hunk 2 above)
Comment (timtay-microsoft) [reply device/iot-device-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/device/DeviceTwin/PropertyTest.java]: It did go into master already.

File: provisioning/security/dice-provider/pom.xml (hunk 3 above)
Comment (timtay-microsoft) [reply provisioning/security/dice-provider/pom.xml]: Which ones did I miss? All artifacts that we publish should have "-preview"

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/Pair.java (hunk 4 above)
Comment (timtay-microsoft) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/Pair.java]: Sure

File: service/iot-service-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/service/devicetwin/PairTest.java (hunk 5 above)
Comment (timtay-microsoft) [reply service/iot-service-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/service/devicetwin/PairTest.java]: Sure

File: versions.json (hunk 6 above)
Comment (timtay-microsoft) [reply versions.json]: David is correct. The "java" and "provisioning" versions belonged to pom files that did not get published as artifacts. There was no reason for us to ever care what version they were, much less to bump them.

**Comment by prmathur-microsoft** [comment]
Please provide details of the failures when ignoring tests
